aiohttp/client.py

```diff
@@ -601,7 +601,9 @@ async def _request(
                     all_cookies = self._cookie_jar.filter_cookies(url)
 
                     if cookies is not None:
-                        tmp_cookie_jar = CookieJar()
+                        tmp_cookie_jar = CookieJar(
+                            quote_cookie=self._cookie_jar.quote_cookie
```

Comment:
Would it make sense to copy the other settings? unsafe and treat_as_secure_origin.

An alternative API might be to add a CookieJar.new() method or something, which simply returns a new CookieJar with the same settings.

Maybe someone else has an opinion on which method to use...

Reply:
I think stdlib has a precedent for `.copy()`. Though, isn't there a `__copy__()`, too?

Reply:
Yeah, it's not a copy though, in that we want an empty CookieJar with the same settings, not one with all the cookies copied over.

Reply:
> I think stdlib has a precedent for `.copy()`. Though, isn't there a `__copy__()`, too?

CookieJar from ClienSession may be DummyCookieJar or user implemented CookieJar. If we use a copy of ClientSession's cookiejar to handle per-request cookie, cookies may not be processed as expected.

Reply:
Right, sounds like we want to avoid copying the class, so probably stick with this approach then.